core-services/workflow/src/main/java/org/silverpeas/core/workflow/engine/instance/ProcessInstanceImpl.java

```diff
@@ -51,170 +63,137 @@
    */
-  private transient HistoryStep currentStep = null;
+  @Transient
+  private HistoryStep currentStep = null;
   /**
    * the status of this instance regarding 'undo' process while true, the atomic operations are not
    * stored anymore
    */
-  private transient boolean inUndoProcess = false;
+  @Transient
+  private boolean inUndoProcess = false;
   /**
-   * Vector of all users who can see this process instance
-   * @field-name interestedUsers
-   * @field-type InterestedUser
-   * @many-key instanceId
-   * @set-method castor_setInterestedUsers
-   * @get-method castor_getInterestedUsers
+   * All users who can see this process instance
    */
-  private Vector<InterestedUser> interestedUsers = null;
+  @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true, mappedBy =
+      "processInstance")
+  private Set<InterestedUser> interestedUsers = new HashSet<>();
   /**
-   * Vector of all users who can act on this process instance
-   * @field-name workingUsers
-   * @field-type WorkingUser
-   * @set-method castor_setWorkingUsers
-   * @get-method castor_getWorkingUsers
+   * All users who can act on this process instance
    */
-  private Vector<WorkingUser> workingUsers = null;
+  @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true, mappedBy =
+      "processInstance")
+  private Set<WorkingUser> workingUsers = new HashSet<>();
   /**
-   * Vector of all users who can have locked a state of this process instance
-   * @field-name lockingUsers
-   * @field-type LockingUser
-   * @many-key instanceId
-   * @set-method castor_setLockingUsers
-   * @get-method castor_getLockingUsers
+   * All users who can have locked a state of this process instance
    */
-  private Vector<LockingUser> lockingUsers = null;
+  @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true, mappedBy =
+      "processInstance")
+  private Set<LockingUser> lockingUsers = new HashSet<>();
   /**
-   * Vector of all states that are due to be resolved for this process instance
-   * @field-name activeStates
-   * @field-type ActiveState
-   * @many-key instanceId
-   * @set-method castor_setActiveStates
-   * @get-method castor_getActiveStates
+   * A1l states that are due to be resolved for this process instance
    */
-  private Vector<ActiveState> activeStates = null;
+  @OneToMany(fetch = FetchType.EAGER, cascade = CascadeType.ALL, orphanRemoval = true, mappedBy =
+      "processInstance")
+  private Set<ActiveState> activeStates = new HashSet<>();
   /**
    * The DataRecord where are stored all the folder fields.
    */
-  private transient DataRecord folder = null;
+  @Transient
+  private DataRecord folder = null;
   /**
    * A Map action -> DataRecord
    */
-  private transient Map<String, DataRecord> actionData = null;
+  @Transient
+  private Map<String, DataRecord> actionData = null;
 
   /**
    * Default constructor
    */
   public ProcessInstanceImpl() {
```

Comment:
![MAJOR](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/severity-major.png 'Severity: MAJOR') Add a nested comment explaining why this method is empty, throw an UnsupportedOperationException or complete the implementation. [![rule](https://raw.githubusercontent.com/SonarCommunity/sonar-github/master/images/rule.png)](http://integration.silverpeas.org/sonar/coding_rules#rule_key=squid%3AS1186)